pyproject.toml

```diff
@@ -3,10 +3,33 @@ requires = [
     "wheel",
     "setuptools<=51.0.0",
     "Cython>=0.29.18",
-    "numpy==1.16.5; python_version=='3.7'",
-    "numpy==1.17.3; python_version=='3.8'",
-    "numpy==1.19.3; python_version=='3.9'",
-    # do not pin numpy on future versions of python to avoid incompatible numpy and python versions
-    "numpy; python_version>='3.10'",
     "pybind11>=2.4.3",
+    "pythran",
```

Comment:
@rgommers hmm, I don't think this belongs in the backport?

Reply:
I've pulled that out now at least, hopefully nothing else subtle in there that shouldn't be

Reply:
indeed, that's correct. sorry forgot to mention; we just need the more elaborate numpy version handling